curriculum/challenges/english/25-front-end-development/lab-real-time-counter/66bb3e20d3dc5b6d0a21f5dd.md

```diff
@@ -0,0 +1,238 @@
+---
+id: 66bb3e20d3dc5b6d0a21f5dd
+title: Build a Real Time Counter
+challengeType: 14
+dashedName: build-a-real-time-counter
+demoType: onClick
+---
+
+# --description--
+
+**Objective:** Fulfill the user stories below and get all the tests to pass to complete the lab. **Do not copy this demo project**.
+
+**User Stories:**
+
+1. You should have an `h1` element with the text `Real-Time Character Counter`.
+1. You should have a `textarea` element with the `id` of `textInput`. It should have a `placeholder` attribute with the text `Type something...`.
+1. There should be a `p` element with the `id` of `charCount`. It should initially contain the text `Character Count: 0/50`. This placeholder text should be replaced with the current count of characters in the `textarea` element.
+1. You should attach an event listener to the `textInput` element that listens for the `input` event. 
+1. Inside the event trigger function, you should get the value of the `textarea` element and store it in the variable `textInput`. You should also declare a variable `charCount` and set it equal to the length of the `textInput` variable.
+1. Create a condition that checks if the current character count is greater than `50`. If it is, the user shouldn't be able to enter more characters. 
+1. Get the character count display element by its ID and update the text to show the current character count using `innerHTML`.
+1. If the character count is equal to `50`, the text should be displayed in `red`. 
+
+# --hints--
+
+There should be an `h1` element with the text of your topic.
+
+```js
+assert.isAbove(document.querySelector('h1')?.innerText.length, 0);
+```
+
+You should have a `textarea` element with the `id` of `textInput`. 
+
+```js
+assert($('textarea').attr('id') == 'textInput');
+```
+
+Your `textarea` should have a `placeholder` attribute with the text `Type something...`.
+    
+```js
+assert($('textarea').attr('placeholder') == 'Type something...');
+```
+
+You should have a `p` element with the `id` of `charCount`. 
+
+```js
+assert($('p').attr('id') == 'charCount');
+```
+
+The `#id` `p` element should initially contain the text `Character Count: 0/50`.
+
+```js
+assert(document.querySelector('p')?.innerText, 'Character Count: 0/50');
+```
+
+You should attach an event listener to the `textInput` element.
+
+```js
+assert.match(code, /document\.getElementById\(\"textInput\"\)\.addEventListener/);
```

Comment:
As Jessica mentioned, one can use `querySelector` instead of `getElementById`. But also, you can get the element and store it in a variable and have `<variableName>.addEventListener`